samtranslator/plugins/application/serverless_app_plugin.py

```diff
@@ -318,8 +329,26 @@ def on_after_transform_template(self, template):
                             TemplateId=self._sanitize_sar_str_param(template_id),
                         )
                     )
-                    response = self._sar_service_call(get_cfn_template, application_id, application_id, template_id)
+                    response = {"Status": "PREPARING"}  # default response if we can't reach SAR
+
+                    try:
+                        if not throttled:
+                            response = self._sar_service_call(
+                                get_cfn_template, application_id, application_id, template_id
+                            )
```

Comment:
since now we manually skip boto3 calls when throttled, it looks like it will go to next iteration immediately, which is likely to result in another throttle. Do you think we should add `sleep` at the end of the iteration if `throttled` is `True`?

Reply:
I should (will?) refactor this code to make it easier to read.  Once `throttled` is `True`, we will continue looping but will not make any more SAR calls.  We continue looping to in order to call `_handle_get_cfn_template_response`.  Once the for loop is done, we will `sleep(self.SLEEP_TIME_SECONDS)` before going through the `while` again, which resets `Throttled`

Reply:
Oh I see, just found the line.